torch_ipex/csrc/cpu/DevOPs.cpp

```diff
@@ -50,16 +50,30 @@ at::Tensor AtenIpexCPUDev::dil_convolution(
   CHECK_DNNL_OP_PRE_COND(weight);
 
   dbl::comm::reorder_to_bf16_for_mix_prec(input);
-  dbl::comm::reorder_to_bf16_for_mix_prec(weight);
-
   dil_input = dbl::comm::try_gen_dil_tensor(input);
-  dil_weight = dbl::comm::try_gen_dil_tensor(weight);
+
   if (bias.defined()) {
     CHECK_DNNL_OP_PRE_COND(bias);
     dbl::comm::reorder_to_bf16_for_mix_prec(bias);
     dil_bias = dbl::comm::try_gen_dil_tensor(bias);
   }
 
+  // Reorder weight
+  auto kdims = weight.dim() - 2;
+  auto stride_vec = dbl::comm::expand_param_if_needed(stride, "stride", kdims);
+  auto padding_vec = dbl::comm::expand_param_if_needed(padding, "padding", kdims);
+  auto dilation_vec = dbl::comm::expand_param_if_needed(dilation, "dilation", kdims);
+  auto packed_desc = dil::convolution_forward::expected_weights_desc(
```

Comment:
If we are querying the format at runtime, we could pass more info to the query API, such as input size
https://github.com/intel/intel-extension-for-pytorch/pull/31/commits/c80d2990b1afa59b36a364c2fa01c57ff50872f3#diff-b3b683f5bad7d86288f07872e911a97eR77